### DISCUSSION TITLE
Is it safe/private to share my password with this application?

### DISCUSSION THREAD

**Discussion by ghost** [icloud-photos-downloader/icloud_photos_downloader#474] [Q&A] [answered]
## Summary
[NOTE]: # ( Provide a brief overview of what the new feature is all about )

Is it safe/private to share my password with this application?
## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )
man privacy! my apple id is my whole life

**Comment by Hermanncillo** [comment]
You can use your original Password/Username and once used, change it :-)

**Comment by kinweilee** [comment]
I have this concern too. Have you tried creating an app-specific password for your apple id?
[https://support.apple.com/en-gb/HT204397](url)

**Comment by boredazfcuk** [comment]
> I have this concern too. Have you tried creating an app-specific password for your apple id?
> [https://support.apple.com/en-gb/HT204397](url)

App specific passwords are for apps that run on iDevices. They are no help here, as you cannot use one to login to iCloud.com via the web.

**Comment by menkej** [comment, marked as answer]
I share your concern. We're using https://pypi.org/project/keyring/ to store the password in the system keyring. The backend is different, depending on the platform you're running this on:
* macOS [Keychain](https://en.wikipedia.org/wiki/Keychain_%28software%29)
* Freedesktop [Secret Service](http://standards.freedesktop.org/secret-service/) supports many DE including GNOME (requires [secretstorage](https://pypi.python.org/pypi/secretstorage))
* KDE4 & KDE5 [KWallet](https://en.wikipedia.org/wiki/KWallet) (requires [dbus](https://pypi.python.org/pypi/dbus-python))
* [Windows Credential Locker](https://docs.microsoft.com/en-us/windows/uwp/security/credential-locker)

If you consider those as safe it should be. ;-) You're free not to safe the password and specify it on each run. But then you could see it in the process list, which might be another issue when working on a shared device...
